src/main/java/com/salesforce/einsteinbot/sdk/client/BasicChatbotClientImpl.java

```diff
@@ -86,11 +96,34 @@ void setHealthApi(HealthApi healthApi) {
     this.healthApi = healthApi;
   }
 
+  @VisibleForTesting
+  void setVersionsApi(VersionsApi versionsApi) {
+    this.versionsApi = versionsApi;
+  }
+
   @Override
   public BotResponse startChatSession(RequestConfig config,
       ExternalSessionId sessionId,
       BotSendMessageRequest botSendMessageRequest) {
 
+    String currentApiVersion;
+    try {
+      currentApiVersion = getCurrentApiVersion();
+      SupportedVersions versions = getSupportedVersions();
+      List<SupportedVersionsVersions> supportedVersions = versions.getVersions()
+          .stream()
+          .filter(v -> {
+            assert v.getVersionNumber() != null;
+            return v.getVersionNumber().equals(currentApiVersion);
+          })
+          .collect(Collectors.toList());
```

Comment:
I don't think you need to collect to list and check size. Instead, can we find first match by using `findFirst()` method in the stream?

Reply:
Yep, makes sense. Updated this